app/controllers/challenges_controller.rb

```diff
@@ -49,28 +51,28 @@ def create
   end
 
   def show
-    @challenge = Challenge.find(params['id']) rescue nil
-    return redirect_to root_path if @challenge.nil?
-
-    # TODO, there is a better way to do this...
-    users = @challenge.entries.map {|e| e.user_id }.uniq
-    @users = User.where(:_id.in => users).inject({}) {|h,u| h[u.id] = u; h}
+    # Limit to id to avoid downloading uneccessary entries
+    challenge_id = params['id']
+    challenge = Challenge.only(:id).find(challenge_id) rescue nil
```

Comment:
If there is no matching challenge, isn't this a no-op? Why do we need to do this query and then query again on line 64 for full set of entries?

Reply:
It is to prevent "format.json" to download full challenge (with all entries) and display only 4 params. I explained it in the commit message https://github.com/igrigorik/vimgolf/pull/234/commits/6ecdc7ec43731aa20d8d312a1bca39aa5b55b618

But still we have a problem in format.html
But it will be solved in another PR that will paginate all the solution.

Reply:
Ah, I see — thank you.